test/404s.test.js

```diff
@@ -1577,6 +1577,36 @@ test('reply.notFound invoked the notFound handler', t => {
   })
 })
 
+test('isNotFound should be present in request object and true if route doesn\'t exist', t => {
+  t.plan(3)
+
+  const fastify = Fastify()
+
+  fastify.addHook('onRequest', (req, reply, done) => {
+    if (req.isNotFound) {
+      reply.code(404).send(new Error('kaboom'))
```

Comment:
Here you should also add a `return` to stop the request lifecycle :)
```suggestion
      return reply.code(404).send(new Error('kaboom'))
```